From 5ec98a06e70c071cc38c45c80804df4afc7eff09 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@redhat.com>
Date: Wed, 22 Jun 2011 14:35:49 -0400
Subject: [PATCH] Fix Ipv4&IPv6 lookup in getaddrinfo

Problem introduced in the last patch.
(cherry picked from commit e12df166d37522c2ed434c2d70a1b04640d2d7c6)

Upstream: yes

---
 sysdeps/posix/getaddrinfo.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sysdeps/posix/getaddrinfo.c b/sysdeps/posix/getaddrinfo.c
index abd8551..6b7423c 100644
--- a/sysdeps/posix/getaddrinfo.c
+++ b/sysdeps/posix/getaddrinfo.c
@@ -904,7 +904,7 @@ gaih_inet (const char *name, const struct gaih_service *service,
 			      pat = &((*pat)->next);
 			      no_data = 0;
 			    }
-			  else if ((*pat)->family == AF_UNSPEC
+			  else if (req->ai_family == AF_UNSPEC
 				   || (*pat)->family == req->ai_family)
 			    {
 			      pat = &((*pat)->next);
-- 
1.7.3.4

